src/libLookup/Lookup.cpp

```diff
@@ -2592,6 +2594,110 @@ bool Lookup::GetOfflineLookupNodes() {
+  if (!Messenger::GetLookupGetDirectoryBlocksFromSeed(message, offset, portNo,
+                                                      index_num)) {
+    LOG_GENERAL(WARNING, "Unable to set GetLookupGetDirectory");
+    return false;
+  }
+
+  vector<unsigned char> msg = {MessageType::LOOKUP,
+                               LookupInstructionType::SETDIRBLOCKSFROMSEED};
+
+  unsigned int curr_offset = MessageOffset::BODY;
+
+  vector<boost::variant<DSBlock, VCBlock, FallbackBlockWShardingStructure>>
+      dirBlocks;
+
+  for (uint64_t i = index_num;
+       i <= m_mediator.m_blocklinkchain.GetLatestIndex(); i++) {
+    BlockLink b = m_mediator.m_blocklinkchain.GetBlockLink(i);
+
+    if (get<BlockLinkIndex::BLOCKTYPE>(b) == BlockType::DS) {
+      dirBlocks.emplace_back(
+          m_mediator.m_dsBlockChain.GetBlock(get<BlockLinkIndex::DSINDEX>(b)));
+    } else if (get<BlockLinkIndex::BLOCKTYPE>(b) == BlockType::VC) {
+      VCBlockSharedPtr vcblockptr;
+      if (!BlockStorage::GetBlockStorage().GetVCBlock(
+              get<BlockLinkIndex::BLOCKHASH>(b), vcblockptr)) {
+        LOG_GENERAL(WARNING, "could not get vc block "
+                                 << get<BlockLinkIndex::BLOCKHASH>(b));
+        continue;
+      }
+      dirBlocks.emplace_back(*vcblockptr);
+    } else if (get<BlockLinkIndex::BLOCKTYPE>(b) == BlockType::FB) {
+      FallbackBlockSharedPtr fallbackwsharding;
+      if (!BlockStorage::GetBlockStorage().GetFallbackBlock(
+              get<BlockLinkIndex::BLOCKHASH>(b), fallbackwsharding)) {
+        LOG_GENERAL(WARNING, "could not get fb block "
+                                 << get<BlockLinkIndex::BLOCKHASH>(b));
+        continue;
+      }
+      dirBlocks.emplace_back(*fallbackwsharding);
+    }
+  }
+
+  if (dirBlocks.empty()) {
+    LOG_GENERAL(WARNING, "No block, aborting..");
+    return false;
+  }
+
+  uint128_t ipAddr = from.m_ipAddress;
+  Peer peer(ipAddr, portNo);
+
+  if (!Messenger::SetLookupSetDirectoryBlocksFromSeed(msg, curr_offset,
+                                                      dirBlocks, index_num)) {
+    LOG_GENERAL(WARNING, "Could not set Directory Blocks");
+    return false;
+  }
+
+  P2PComm::GetInstance().SendMessage(peer, msg);
+
+  return true;
+}
+
+bool Lookup::ProcessSetDirectoryBlocksFromSeed(
+    const vector<unsigned char>& message, unsigned int offset,
+    [[gnu::unused]] const Peer& from) {
+  vector<boost::variant<DSBlock, VCBlock, FallbackBlockWShardingStructure>>
+      dirBlocks;
+  uint64_t index_num;
+
+  if (!Messenger::GetLookupSetDirectoryBlocksFromSeed(message, offset,
+                                                      dirBlocks, index_num)) {
+    LOG_GENERAL(WARNING, "Could not Get SetDirectoryBlocksFromSeed");
+    return false;
+  }
+
+  if (dirBlocks.empty()) {
+    LOG_GENERAL(WARNING, "No Directory blocks sent");
+    return false;
+  }
+
+  // do something
+
+  return true;
+}
+
+void Lookup::ComposeGetDirectoryBlocksFromSeed(uint64_t& index_num) {
```

Comment:
I think it should say `ComposeSet...` since you are composing the set message. But since you are also sending it, might be better to rename as `ComposeAndSendSet...`

Reply:
actually it is wrong inside the msg instruction, should be `GETDIRECTORY..`